runtime/src/core/tags.js

```diff
@@ -0,0 +1,52 @@
+import { memory, view } from '../runtime';
+import { throwGrainError } from '../errors/errors';
+import {
+  GRAIN_ERR_NOT_NUMBER_GENERIC,
+  GRAIN_ERR_NOT_BOOLEAN_GENERIC,
+  GRAIN_ERR_NOT_TUPLE_GENERIC,
+  GRAIN_ERR_NOT_LAMBDA_GENERIC,
+  GRAIN_ERR_NOT_STRING_GENERIC,
+  GRAIN_ERR_NOT_DOM_ELEMENT_GENERIC
+} from '../errors/error-codes';
+
+export const GRAIN_NUMBER_TAG_MASK = 0b0001;
```

Comment:
Maybe we should change these to a more OO-style?
Something like
```js
class GrainTag {
  constructor(code) {
    this.code = code;
  }
  // ...
}
```
(not just here, but other places as well...it's kind of obvious we ported a lot of this from C)

Reply:
I agree that this can be better, but I feel like that's out of scope for what I was trying to accomplish with this. We should make an issue to tackle that.

Reply:
Yes, that's what I meant; I'm not asking for you to change it for this PR.